tomni/cytosmart_data_format/main.py

```diff
@@ -196,6 +196,33 @@ def delete_annotation(self, item: Annotation):
         """
         pass
 
+    def filter(
+        self, feature: str, min_val: float, max_val: float, inplace: bool = False,
+    ):
+        """Filter annotations by feature.
+
+        Args:
+            feature (str): Feature name, i.e. `roundness` or `area`.
+            min_val (float): Minimum value to threshold.
+            max_val (float): Maximum value to threshold
+            inplace (bool, optional): If True, filter in-place. Modifies the object internally. If False, return collection of annotations. Defaults to False.
+
+        Returns:
+            CytoSmartDataFormat or List[Annotation]: Collection of filtered annotions or if `inplace=True` object with filterd annotations.
+        """
+        filtered_annotations = []
+
+        for annotation in self._annotations:
+            feature_value = getattr(annotation, feature)
+            if min_val <= feature_value <= max_val:
+                filtered_annotations.append(annotation)
+
+        if inplace:
+            self._annotations = filtered_annotations
+            return self
+
+        return filtered_annotations
```

Comment:
very cool, but please add unittest because now we just assume it is working

Reply:
> very cool, but please add unittest because now we just assume it is working

Now it is even cooler with unit tests.